src/main/java/build/buildfarm/worker/InputFetcher.java

```diff
@@ -307,4 +309,24 @@ public void run() {
       }
     }
   }
+
+  private void failOperation(String failureMessage, String failureDetails)
+      throws InterruptedException {
+    Operation failedOperation =
+        FailedOperationGetter.get(
+            operationContext.operation,
+            operationContext.queueEntry.getExecuteEntry(),
+            failureMessage,
+            failureDetails);
+
+    try {
+      workerContext.putOperation(failedOperation);
```

Comment:
afaict, the error path removes the operation and expires the poller so that the operation gets requeued.  It wasn't until I put the operation into the context that I was able to inform the client about the execdir failures.